common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -71,226 +76,219 @@ private static void captureIde(BuildScanExtension buildScan, ProviderFactory pro
-        return envVariable("JENKINS_URL", providers).isPresent();
+    private boolean isJenkins() {
+        return envVariable("JENKINS_URL").isPresent();
     }
 
-    private static boolean isHudson(ProviderFactory providers) {
-        return envVariable("HUDSON_URL", providers).isPresent();
+    private boolean isHudson() {
+        return envVariable("HUDSON_URL").isPresent();
     }
 
-    private static boolean isTeamCity(ProviderFactory providers) {
-        return envVariable("TEAMCITY_VERSION", providers).isPresent();
+    private boolean isTeamCity() {
+        return envVariable("TEAMCITY_VERSION").isPresent();
     }
 
-    private static boolean isCircleCI(ProviderFactory providers) {
-        return envVariable("CIRCLE_BUILD_URL", providers).isPresent();
+    private boolean isCircleCI() {
+        return envVariable("CIRCLE_BUILD_URL").isPresent();
     }
 
-    private static boolean isBamboo(ProviderFactory providers) {
-        return envVariable("bamboo_resultsUrl", providers).isPresent();
+    private boolean isBamboo() {
+        return envVariable("bamboo_resultsUrl").isPresent();
     }
 
-    private static boolean isGitHubActions(ProviderFactory providers) {
-        return envVariable("GITHUB_ACTIONS", providers).isPresent();
+    private boolean isGitHubActions() {
+        return envVariable("GITHUB_ACTIONS").isPresent();
     }
 
-    private static boolean isGitLab(ProviderFactory providers) {
-        return envVariable("GITLAB_CI", providers).isPresent();
+    private boolean isGitLab() {
+        return envVariable("GITLAB_CI").isPresent();
     }
 
-    private static boolean isTravis(ProviderFactory providers) {
-        return envVariable("TRAVIS_JOB_ID", providers).isPresent();
+    private boolean isTravis() {
+        return envVariable("TRAVIS_JOB_ID").isPresent();
     }
 
-    private static boolean isBitrise(ProviderFactory providers) {
-        return envVariable("BITRISE_BUILD_URL", providers).isPresent();
+    private boolean isBitrise() {
+        return envVariable("BITRISE_BUILD_URL").isPresent();
     }
 
-    private static void captureGitMetadata(BuildScanExtension buildScan) {
-        buildScan.background(api -> {
-            if (!isGitInstalled()) {
-                return;
-            }
+    private void captureGitMetadata() {
+        buildScan.background(captureGitMetadataAction);
+    }
 
-            String gitRepo = execAndGetStdOut("git", "config", "--get", "remote.origin.url");
-            String gitCommitId = execAndGetStdOut("git", "rev-parse", "--verify", "HEAD");
-            String gitCommitShortId = execAndGetStdOut("git", "rev-parse", "--short=8", "--verify", "HEAD");
-            String gitBranchName = execAndGetStdOut("git", "rev-parse", "--abbrev-ref", "HEAD");
-            String gitStatus = execAndGetStdOut("git", "status", "--porcelain");
+    private static final Action<BuildScanExtension> captureGitMetadataAction =
```

Comment:
ah i see, writing it in this way ensure that it doesn't capture the outer class and ensure it is compatible with the configuration cache. nice find. i wonder if we should document this in the code to avoid future incompatible changes or write a test for it.

Reply:
Yes, that was the plan.